scripts/core/superdesk-api.d.ts

```diff
@@ -84,6 +91,7 @@ declare module 'superdesk-api' {
             articleGridItemWidgets?: Array<React.ComponentType<{article: IArticle}>>;
             authoringTopbarWidgets?: Array<React.ComponentType<{article: IArticle}>>;
             authoringSideWidgets?: Array<IAuthoringSideWidget>;
+            authoringHeaderComponents?: Array<AuthoringHeaderItem>;
```

Comment:
Can you use `customFieldTypes` instead? We shouldn't add multiple API extension points when one already exists that does 90% of what's needed. The problem with `customFieldTypes` in this case is that they have to be configured via content profiles. We could add `always-on` option to the interface and the same extension point should work then.

```
export interface ICustomFieldType<IConfig> {
        id: string;
        label: string;
        editorComponent: React.ComponentType<IEditorComponentProps<IConfig>>;
        previewComponent: React.ComponentType<IPreviewComponentProps>;
        configComponent?: React.ComponentType<IConfigComponentProps<IConfig>>;

        /**
         * 'content-profiles'   - field will only appear in authoring when configured in content profile.
         * 'always-on'          - field will always appear in authoring.
         *                      configuration via content profiles will not be available.
         */
        configuration?: 'content-profiles' | 'always-on'; // defaults to 'content-profiles'
    }
```

Reply:
but how to specify component that will be in header? Extend it? and add `headerComponent`?

Reply:
You would need to provide `editorComponent` and inside the extension you would configure it to be in read-only mode. Components defined in `previewComponent` are only rendered in the [preview pane ](https://user-images.githubusercontent.com/2076953/138265289-f1080998-ce3a-4d7a-b49b-04e2c03700fb.png)